app/controllers/FurloughPeriodQuestionController.scala

```diff
@@ -46,49 +48,51 @@ class FurloughPeriodQuestionController @Inject()(
-            case Some(furloughEnd) => Future.successful(Ok(view(preparedForm, furloughStart, furloughStatus, Some(furloughEnd))))
-            case None =>
-              Logger.error("expecting FurloughEndDate in mongo when its furlough ended, but not found")
-              Future.successful(Redirect(routes.ErrorController.somethingWentWrong()))
-          }
+        furloughStatus match {
+          case FurloughOngoing => Future.successful(Ok(view(preparedForm, furloughStart, furloughStatus, None)))
+          case FurloughEnded =>
+            getAnswer(FurloughEndDatePage) match {
+              case Some(furloughEnd) => Future.successful(Ok(view(preparedForm, furloughStart, furloughStatus, Some(furloughEnd))))
+              case None =>
+                Logger.error("expecting FurloughEndDate in mongo when its furlough ended, but not found")
+                Future.successful(Redirect(routes.ErrorController.somethingWentWrong()))
+            }
+        }
       }
-    }
   }
 
-  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    getRequiredAnswers(FurloughStartDatePage, FurloughStatusPage) { (furloughStart, furloughStatus) =>
-      form
-        .bindFromRequest()
-        .fold(
-          formWithErrors =>
-            furloughStatus match {
-              case FurloughOngoing => Future.successful(BadRequest(view(formWithErrors, furloughStart, furloughStatus, None)))
-              case FurloughEnded =>
-                getAnswer(FurloughEndDatePage) match {
-                  case Some(furloughEnd) =>
-                    Future.successful(BadRequest(view(formWithErrors, furloughStart, furloughStatus, Some(furloughEnd))))
-                  case None =>
-                    Logger.error("expecting FurloughEndDate in mongo when its furlough ended, but not found")
-                    Future.successful(Redirect(routes.ErrorController.somethingWentWrong()))
-                }
-          },
-          value =>
-            for {
-              updatedAnswers <- Future.fromTry(request.userAnswers.set(FurloughPeriodQuestionPage, value))
-              _              <- sessionRepository.set(updatedAnswers)
-            } yield Redirect(navigator.nextPage(FurloughPeriodQuestionPage, updatedAnswers))
-        )
-    }
+  def onSubmit(): Action[AnyContent] = (identify andThen feature(FastTrackJourneyFlag) andThen getData andThen requireData).async {
+    implicit request =>
+      getRequiredAnswers(FurloughStartDatePage, FurloughStatusPage) { (furloughStart, furloughStatus) =>
+        form
+          .bindFromRequest()
+          .fold(
+            formWithErrors =>
+              furloughStatus match {
+                case FurloughOngoing => Future.successful(BadRequest(view(formWithErrors, furloughStart, furloughStatus, None)))
+                case FurloughEnded =>
+                  getAnswer(FurloughEndDatePage) match {
```

Comment:
same as above

Reply:
i can do 
```if(furloughStatus == FurloughOngoing) {
          Future.successful(Ok(view(preparedForm, furloughStart, furloughStatus, None)))
        } else {
          getAnswer(FurloughEndDatePage) match {
            case Some(furloughEnd) => Future.successful(Ok(view(preparedForm, furloughStart, furloughStatus, Some(furloughEnd))))
            case None =>
              Logger.error("expecting FurloughEndDate in mongo when its furlough ended, but not found")
              Future.successful(Redirect(routes.ErrorController.somethingWentWrong()))
          }
        }``` 
but it doesn't really make any difference , not sure how i can make it better